File: Backend/src/main/java/com/fruitshop/backend/service/WishlistService.java
package com.fruitshop.backend.service;

import com.fruitshop.backend.dto.ApiResponse;
import com.fruitshop.backend.dto.WishlistResponseDto;

import java.util.List;

public interface WishlistService {
    ApiResponse<String> addToWishlist(Integer userId, Integer productId);
    ApiResponse<String> removeFromWishlist(Integer userId, Integer productId);
    ApiResponse<List<WishlistResponseDto>> getWishlist(Integer userId);
}

File: Frontend/src/services/wishlistService.ts
import { callApiWithMethod, type ApiResponse } from "../utils/apiClient";

export interface WishlistResponseDto {
  wishlistId: number;
  productId: number;
  productName: string;
  price: number;
  imageUrl: string;
  categoryName: string;
}

export const getUserWishlist = async (userId: number): Promise<ApiResponse<WishlistResponseDto[]>> => {
  return await callApiWithMethod<never, ApiResponse<WishlistResponseDto[]>>("GET", `/api/wishlist/${userId}`);
};

export const addToWishlist = async (userId: number, productId: number): Promise<ApiResponse<string>> => {
  return await callApiWithMethod<never, ApiResponse<string>>("POST", `/api/wishlist/${userId}/add/${productId}`);
};

export const removeFromWishlist = async (userId: number, productId: number): Promise<ApiResponse<string>> => {
  return await callApiWithMethod<never, ApiResponse<string>>("DELETE", `/api/wishlist/${userId}/remove/${productId}`);
};
